crypto/fipsmodule/service_indicator/service_indicator.c

```diff
@@ -346,24 +346,60 @@ void HMAC_verify_service_indicator(const EVP_MD *evp_md) {
   }
 }
 
-void PBKDF_verify_service_indicator(const EVP_MD *evp_md) {
+void HKDF_verify_service_indicator(const EVP_MD *evp_md,
+  const uint8_t *salt, size_t salt_len) {
+  // HKDF with SHA1, SHA224, SHA256, SHA384, and SHA512 are approved.
+  //
+  // FIPS 140 parameter requirements, per NIST SP 800-108 Rev. 1:
+  //
+  // It is recommended that the length of a KDK used by a KDF be at least as
+  // large as the targeted security strength (in bits) of any application that
+  // will be supported by the use of the derived keying material.
+  //
+  // We can't test for that, as we don't know what the HKDF output will be
+  // used for.
+  //
+  // Per SP800-56Crev2, the salt cannot be empty in FIPS mode; a NULL salt
+  // is replaced in the HMAC with a buffer filled with NUL bytes (0x00), so
+  // that's OK.
```

Comment:
```suggestion
  // is replaced in the HMAC with a buffer filled with NUL bytes (0x00), 
  // when |evp_md| is not NULL, so that's OK.
  // Note that we reach this check if |HMAC| succeeds which cannot be 
  // the case if |evp_md| = NULL; it would have then failed in |HMAC_Init_ex|.
```